elastic/elastic-agent
# mage: check that links are live in the OTel README

This PR adds a utility based on the `link-patrol` command to check if links in a given file are live. This utility should be reusable so we can use it to validate other files in the repository. 

Also added a check in the `check` and `otel:readme` mage targets to validate all links in the Otel Readme are valid.

## What does this PR do?



## Why is it important?


## How to test this PR locally

You can either run `mage check` or `mage otel:readme` to trigger the link check. 

To test it indeed fails if a link is invalid you can introduce a typo and it should fail in subsequent runs.

`sed -i 's/featuregate/fff/' internal/pkg/otel/templates/README.md.tmpl` 

## Related issues

- Closes #5117
